### ISSUE TITLE
 I love this, could we colaborate? I maintain https://github.com/xplshn/bigdl, currently, there are 1550 (statically linked) binaries in the repos of my package manager #30 

### ISSUE THREAD

**Issue by xplshn** [stal-ix/ix#454]
BigDL is a "package" manager (not really, it manager binaries), its quite simple and small, I'd love us to collaborate (making universal, single file binaries, statically linked, against Musl if possible). I know this could well be considered spam, but I think both projects would benefit a lot from this!

Check out BigDL here: https://github.com/xplshn/bigdl

This is the help page of `bigdl`:
```
]@ bigdl -h
Usage:
 [-v|-h] [list|install|remove|update|run|info|search|tldr] <-args->

Options:
 -h, --help       Show this help message
 -v, --version    Show the version number

Commands:
 list             List all available binaries
 install, add     Install a binary
 remove, del      Remove a binary
 update           Update binaries, by checking their SHA against the repo's SHA
 run              Run a specified binary from cache
 info             Show information about a specific binary OR display installed binaries
 search           Search for a binary - (not all binaries have metadata. Use list to see all binaries)
 tldr             Equivalent to "run --transparent --verbose tlrc" as argument

Variables:
 BIGDL_PRBAR      If present, and set to ZERO (0), the download progressbar will be disabled
 BIGDL_TRUNCATION If present, and set to ZERO (0), string truncation will be disabled
 BIGDL_CACHEDIR   If present, it must contain a valid directory
 INSTALL_DIR      If present, it must contain a valid directory

Examples:
 bigdl search editor
 bigdl install micro
 bigdl install lux kakoune aretext shfmt
 bigdl install --silent bed && echo "[bed] was installed to $INSTALL_DIR/bed"
 bigdl del bed
 bigdl del orbiton tgpt lux
 bigdl info
 bigdl info jq
 bigdl list --described
 bigdl tldr gum
 bigdl run --verbose curl -qsfSL "https://raw.githubusercontent.com/xplshn/bigdl/master/stubdl" | sh -
 bigdl run --silent elinks -no-home "https://fatbuffalo.neocities.org/def"
 bigdl run --transparent --silent micro ~/.profile
 bigdl run btop

Version: 1.6.9p
```

**Comment by pg83** [comment]
>I'd love us to collaborate

Sounds good. 

I think that you can use IX as a provider of binaries for your package manager.

You can run it on your CI, or I can give access to my build farm, with prebuild binaries. Also you can help me with new recipes of not yet packaged binaries.

Any other ideas?

Also you can join us at https://t.me/stal_ix

**Comment by xplshn** [comment]
These is where I pull the binaries from, I don't maintain the repo. Its all automated:
- https://github.com/Azathothas/Toolpacks [https://bin.ajam.dev/]
- https://github.com/Azathothas/Static-Binaries [https://bin.ajam.dev/*/Baseutils/]

So, I wanted to know where I can download the compiled binaries of IX. Also, there isn't a lot of documentation about how IX works.

>Also you can help me with new recipes of not yet packaged binaries.

Sure! If I can actually get some programs to compile statically I will surely hit you up, I'd really like to expand `bigdl` and make it become a viable alternative as a "package" manager.

**Comment by xplshn** [comment]
If you could instruct me on how to work with IX's repos. I think I could do a good rewrite of the *.py scripts into Go. (I am not really sure which of those .py is the client, or if they are all to build the binaries)

**Comment by pg83** [comment]
> good rewrite of the *.py scripts

which scripts exactly?

I have a execution graph building engine in python, but it is really tightly coupled with jinja2 templates, which renders actual build scripts, so, this will be very hard.

**Comment by pg83** [comment]
> So, I wanted to know where I can download the compiled binaries of IX. 

You can't right now, because IX is source based repo, as Gentoo.

> Also, there isn't a lot of documentation about how IX works.

Understand. Which part of documentation I should fill right now? :) https://stal-ix.github.io/IX.html

**Comment by xplshn** [comment]
I didn't see that link. Sorry, my bad. Then, seems like everything's good.
I'm thinking that maybe you could automate builds using Github Actions same way this repo here works: https://github.com/Azathothas/Toolpacks

>You can't right now

I'd be grateful if you'd hit me up to add support for IX in `bigdl` once you have figured out how to do binary distribution
